src/precice/bindings/matlab/+precice/@SolverInterface/SolverInterface.m

```diff
@@ -0,0 +1,283 @@
+            preciceGateway(uint8(53),int32(meshID),int32(firstVertexID),int32(secondVertexID),int32(thirdVertexID),int32(fourthVertexID));
+        end
+        
+        %% Data Access
+        % hasDataID
+        function bool = hasData(obj,dataName,meshID)
+            if ischar(dataName)
+                dataName = string(dataName);
+            end
+            bool = preciceGateway(uint8(60),dataName,int32(meshID));
+        end
+        
+        % getDataID
+        function id = getDataID(obj,dataName,meshID)
+            if ischar(dataName)
+                dataName = string(dataName);
+            end
+            id = preciceGateway(uint8(61),dataName,int32(meshID));
+        end
+        
+        % mapReadDataTo
+        function mapReadDataTo(obj,meshID)
+            preciceGateway(uint8(62),int32(meshID));
+        end
+        
+        % mapWriteDataFrom
+        function mapWriteDataFrom(obj,meshID)
+            preciceGateway(uint8(63),int32(meshID));
+        end
+        
+        % writeBlockVectorData
+        function writeBlockVectorData(obj,dataID,inSize,valueIndices,values)
+            if ~isa(valueIndices,'int32')
+                warning('valueIndices should be allocated as int32 to prevent copying.');
+                valueIndices = int32(valueIndices);
+            end
+            preciceGateway(uint8(64),int32(dataID),int32(inSize),valueIndices,values);
+        end
+        
+        % writeVectorData
+        function writeVectorData(obj,dataID,valueIndex,value)
+            preciceGateway(uint8(65),int32(dataID),int32(valueIndex),value);
+        end
+        
+        % writeBlockScalarData
+        function writeBlockScalarData(obj,dataID,inSize,valueIndices,values)
+            if ~isa(valueIndices,'int32')
+                warning('valueIndices should be allocated as int32 to prevent copying.');
+                valueIndices = int32(valueIndices);
+            end
+            preciceGateway(uint8(66),int32(dataID),int32(inSize),valueIndices,values);
+        end
+        
+        % writeScalarData
+        function writeScalarData(obj,dataID,valueIndex,value)
+            preciceGateway(uint8(67),int32(dataID),int32(valueIndex),value);
+        end
+        
+        % readBlockVectorData
+        function values = readBlockVectorData(obj,dataID,inSize,valueIndices)
+            if ~isa(valueIndices,'int32')
+                warning('valueIndices should be allocated as int32 to prevent copying.');
+                valueIndices = int32(valueIndices);
+            end
+            values = preciceGateway(uint8(68),int32(dataID),int32(inSize),valueIndices);
+        end
+        
+        % readVectorData
+        function value = readVectorData(obj,dataID,valueIndex)
+            value = preciceGateway(uint8(69),int32(dataID),int32(valueIndex));
+        end
+        
+        % readBlockScalarData
+        function values = readBlockScalarData(obj,dataID,inSize,valueIndices,transpose)
```

Comment:
We could try to get rid of `inSize` by directly extracting the size from `size(valueIndices)`.